src/FormMigration/Steps/MailchimpSettings.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Give\FormMigration\Steps;
+
+use Give\FormMigration\Contracts\FormMigrationStep;
+use Give\Framework\Blocks\BlockModel;
+
+/**
+ * @unreleased
+ */
+class MailchimpSettings extends FormMigrationStep
+{
+    /**
+     * @unreleased
+     */
+    public function process(): void
+    {
+        $prevFormId = $this->formV2->id;
+
+        $isFormEnabled = $this->getFormMetaValue($prevFormId, '_give_mailchimp_enable') === 'true';
+        $isFormDisabled = $this->getFormMetaValue($prevFormId, '_give_mailchimp_disable') === 'true';
+        $isGloballyEnabled = give_get_option( 'give_mailchimp_show_checkout_signup') === 'on';
+
+        if ($isFormDisabled || (!$isGloballyEnabled && !$isFormEnabled)) {
```

Comment:
I would suggest moving the "if this should migrate" login in the `canHandle` method of the migration step, which is an optional method that runs before the migration step is processed. By default it just returns true, but you can define the method and add your logic there.

The result is a `process` method which only does processing, like making the block and inserting the block.

Reply:
Much cleaner. a0161125118cc5af042dc2b0087be48f697359db